src/Plugin.cpp

```diff
@@ -92,97 +99,122 @@ void Plugin::init() {
     mInputManager->registerSelectable(body);
 
     body->setSun(mSolarSystem->getSun());
-    mSimpleBodyNodes.push_back(mSceneGraph->NewOpenGLNode(mSceneGraph->GetRoot(), body.get()));
-    mSimpleBodies.push_back(body);
+    auto parent = mSceneGraph->NewOpenGLNode(mSceneGraph->GetRoot(), body.get());
+    mSimpleWMSBodyNodes.push_back(parent);
+    mSimpleWMSBodies.push_back(body);
+
+    VistaOpenSGMaterialTools::SetSortKeyOnSubtree(
+        parent, static_cast<int>(cs::utils::DrawOrder::ePlanets));
   }
-  mSolarSystem->pActiveBody.onChange().connect(
+
+  mActiveBodyConnection = mSolarSystem->pActiveBody.connectAndTouch(
       [this](std::shared_ptr<cs::scene::CelestialBody> const& body) {
-        auto simpleBody = std::dynamic_pointer_cast<SimpleBody>(body);
+        auto simpleWMSBody = std::dynamic_pointer_cast<SimpleWMSBody>(body);
 
-        if (!simpleBody) {
+        // Remove time intervalls of the old body.
+        removeTimeIntervall(mIntervalsOnTimeline);
+
+        if (!simpleWMSBody) {
           return;
         }
-        removeTimeIntervall(mIntervalsOnTimeline);
-        mGuiManager->getSideBar()->callJavascript("clear_container", "set_wms");
-        for (auto const& wms : simpleBody->getWms()) {
-          bool active = wms.mName == simpleBody->getActiveWms().mName;
-          mGuiManager->getSideBar()->callJavascript(
-              "add_dropdown_value", "set_wms", wms.mName, wms.mName, active);
-          if(active) {
-            mIntervalsOnTimeline = simpleBody->getTimeIntervals();
-            addTimeIntervall(mIntervalsOnTimeline);
+
+        mGuiManager->getGui()->callJavascript("CosmoScout.gui.clearDropdown", "wms.setTilesImg");
+        for (auto const& wms : simpleWMSBody->getWms()) {
+          bool active = wms.mName == simpleWMSBody->getActiveWms().mName;
+          mGuiManager->getGui()->callJavascript(
+              "CosmoScout.gui.addDropdownValue", "wms.setTilesImg", wms.mName, wms.mName, active);
+          if (active) {
+            mGuiManager->getGui()->callJavascript(
+                "CosmoScout.simpleWMSBodies.setWMSDataCopyright", wms.mCopyright);
+
+            mIntervalsOnTimeline = simpleWMSBody->getTimeIntervals();
+            addTimeIntervall(mIntervalsOnTimeline, wms.mName, simpleWMSBody->getCenterName());
           }
         }
       });
 
-  mGuiManager->getSideBar()->registerCallback<std::string>(
-        "set_wms", ([this](std::string const& name) {
-          auto body = std::dynamic_pointer_cast<SimpleBody>(mSolarSystem->pActiveBody.get());
-          if (body) {
-            removeTimeIntervall(mIntervalsOnTimeline);
-            body->setActiveWms(name);
-            mIntervalsOnTimeline = body->getTimeIntervals();
-            addTimeIntervall(mIntervalsOnTimeline);
-          }
-        }));
+  mGuiManager->getGui()->registerCallback("wms.setTilesImg",
+      "Set the current planet's wms channel to the TileSource with the given name.",
```

Comment:
I think this callback name is a copy-paste issue, as it has nothing to do with tiles. It could rather be "simpleWmsBodies.setWMS".

```suggestion
  mGuiManager->getGui()->registerCallback("simpleWmsBodies.setWMS",
      "Set the current planet's WMS source to the one with the given name.",
```